src/geometry/rtree.cpp

```diff
@@ -88,36 +90,39 @@ auto from_wkt(std::string_view wkt) -> geom_type {
 template <typename cs, typename query_results = RTree<cs>::query_results>
 auto generate_query_iterator(query_results&& results, std::size_t& alloc) -> IndexIterator* {
   auto geometry_query_iterator = new (alloc) QueryIterator{
-      results | std::views::transform([](auto&& doc) -> t_docId { return get_id<cs>(doc); }),
+      results | std::views::transform([](auto const& doc) -> t_docId { return get_id<cs>(doc); }),
       alloc};
   return geometry_query_iterator->base();
 }
 
 template <typename cs>
-constexpr auto geometry_reporter = [](auto&& geom) -> std::size_t {
+constexpr auto geometry_reporter = [](auto const& geom) -> std::size_t {
   using point_type = typename RTree<cs>::point_type;
   if constexpr (std::is_same_v<point_type, std::decay_t<decltype(geom)>>) {
     return 0ul;
   } else {
     auto const& inners = geom.inners();
     auto outer_size = geom.outer().get_allocator().report();
     return std::accumulate(inners.begin(), inners.end(), outer_size,
-                           [](std::size_t acc, auto&& hole) -> std::size_t {
+                           [](std::size_t acc, auto const& hole) -> std::size_t {
                              return acc + hole.get_allocator().report();
                            });
   }
 };
 
 template <typename cs>
-constexpr auto filter_results = [](auto&& geom1, auto&& geom2) -> bool {
+constexpr auto within_filter = [](auto const& geom1, auto const& geom2) -> bool {
   using point_type = typename RTree<cs>::point_type;
   if constexpr (std::is_same_v<point_type, std::decay_t<decltype(geom2)>> &&
                 !std::is_same_v<point_type, std::decay_t<decltype(geom1)>>) {
     return false;
   } else {
-    return !bg::within(geom1, geom2);
+    return bg::within(geom1, geom2);
```

Comment:
Explanation: filtering is done before-hand and not removing from the matching results afterwards